File: C# Advanced/Sets And Dictionaries Exercises/Phonebook/Phonebook.cs
using System;
using System.Collections.Generic;

namespace Phonebook
{
    public class Phonebook
    {
        public static void Main()
        {
            var phonebook = new Dictionary<string, string>();

            while (true)
            {
                var input = Console.ReadLine();

                if (input.Equals("search"))
                {
                    break;
                }
                else
                {
                    var tokens = input.Split('-');
                    var name = tokens[0];
                    var number = tokens[1];

                    phonebook[name] = number;
                }
            }

            while (true)
            {
                var input = Console.ReadLine();

                if (input.Equals("stop"))
                {
                    break;
                }
                else
                {
                    if (phonebook.ContainsKey(input))
                    {
                        Console.WriteLine($"{input} -> {phonebook[input]}");
                    }
                    else
                    {
                        Console.WriteLine($"Contact {input} does not exist.");
                    }
                }
            }
        }
    }
}

File: Software Technologies/Java/src/Phonebook.java
import java.util.*;

public class Phonebook {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        HashMap<String, String> phonebook = new HashMap<>();

        while (true) {
            String input = sc.nextLine();

            if (input.equals("END")){
                break;
            }

            String[] commands = input.split("\\s+");

            String command = commands[0];
            String name = commands[1];

            if (command.equals("A")){
                String phoneNumber = commands[2];

                if (!phonebook.containsKey(name)){
                    phonebook.put(name, phoneNumber);
                }

                phonebook.put(name, phoneNumber);
            } else {
                if (!phonebook.containsKey(name)){
                    System.out.printf("Contact %s does not exist.", name);
                    System.out.println();
                } else {
                    System.out.printf("%s -> %s", name, phonebook.get(name));
                    System.out.println();
                }
            }
        }
    }
}
